saleor/dashboard/organization/views.py

```diff
@@ -0,0 +1,115 @@
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.decorators import permission_required
+from django.db.models import Q
+from django.http import JsonResponse
+from django.shortcuts import get_object_or_404, redirect
+from django.template.response import TemplateResponse
+from django.utils.translation import pgettext_lazy
+
+from ...account.models import Organization, User
+from ...core.utils import get_paginator_items
+from ...order.models import Order
+from ..views import staff_member_required
+from .filters import OrganizationFilter
+from .forms import OrganizationDeleteForm, OrganizationForm
+
+
+@staff_member_required
+@permission_required('account.manage_organizations')
+def organization_list(request):
+    organizations = (
+        Organization.objects
+        .select_related('default_billing_address')
+        .order_by('name'))
+    organization_filter = OrganizationFilter(request.GET, queryset=organizations)
+    organizations = get_paginator_items(
+        organization_filter.qs, settings.DASHBOARD_PAGINATE_BY,
+        request.GET.get('page'))
+    ctx = {
+        'organizations': organizations, 'filter_set': organization_filter,
+        'is_empty': not organization_filter.queryset.exists()}
+    return TemplateResponse(request, 'dashboard/organization/list.html', ctx)
+
+
+@staff_member_required
+@permission_required('account.manage_organizations')
+def organization_details(request, pk):
+    queryset = Organization.objects.select_related('default_billing_address')
+    organization = get_object_or_404(queryset, pk=pk)
+    organization_orders = Order.objects.filter(user__organization=organization)
```

Comment:
For my use case, this is desired. I think this would be generally desired for a company. If I added a new purchaser to my organization, I'd want them to be able to look at past purchases.

Reply:
Then I'd customize this in your fork, but as a default behavior in Saleor this is not desired. Another edge-case: a user belongs to the Organization A, he places some orders, then he is reassigned to the Company B. Orders of Company A become visible for Company B. I think we should add a relationship between Order and Organization to be able to determine where the orders really belong.